docs/zh/design/api_plugin/design.md

```diff
@@ -1,54 +1,65 @@
 # API 插件设计文档 & 使用指南
+
 本文前半部分讨论 API 插件解决什么问题、为什么这样设计，后半部分介绍 API 插件怎么用。
 
 如果您只关心 API 插件的用法，可以直接跳到 [2.4. 使用指南](zh/design/api_plugin/design?id=_24-使用指南)
 
 ## 1. 需求分析
+
 ### 1.1. 解决什么问题
-解决扩展性问题。不管是Dapr还是开源Layotto的API，目前都无法完全满足生产需求。
 
-回看操作系统领域POSIX API和system call的发展历史，我们可以学到很多，借此预测Runtime的未来。我们可以说，Runtime API将来也不可能完全满足用户需求。想想OS领域，即使有POSIX API了，一些场景还是需要绕开标准API、用特殊指令操作特殊硬件。
+解决扩展性问题。不管是 Dapr 还是开源 Layotto 的 API，目前都无法完全满足生产需求。
 
+回看操作系统领域 POSIX API 和 system call 的发展历史，我们可以学到很多，借此预测 Runtime 的未来。我们可以说，Runtime API 将来也不可能完全满足用户需求。想想 OS 领域，即使有 POSIX API 了，一些场景还是需要绕开标准
+API、用特殊指令操作特殊硬件。
 
-Dapr的扩展性是通过Binding API解决，但是这种非结构化的API有很多问题（比如破坏可移植性、不支持stream等语义）
+Dapr 的扩展性是通过 Binding API 解决，但是这种非结构化的 API 有很多问题（比如破坏可移植性、不支持 stream 等语义）
 
 ### 1.2. 用户场景和需求
+
 举例来说，有以下用户场景：
-1. 公司有自己的定制API需求，因为是非通用需求、不适合做到开源Layotto/Dapr上，于是公司的中间件团队想自己开发到sidecar里。如果公司的项目import 开源Layotto或者Dapr，按目前的架构是没法扩展开发API的，只能Fork出来做扩展
+
+1. 公司有自己的定制 API 需求，因为是非通用需求、不适合做到开源 Layotto/Dapr 上，于是公司的中间件团队想自己开发到 sidecar 里。如果公司的项目 import 开源 Layotto 或者 Dapr，按目前的架构是没法扩展开发 API 的，只能 Fork
+   出来做扩展
 
 ![image](https://user-images.githubusercontent.com/26001097/131614836-60d797c8-b80b-4018-ad43-c2b874d35660.png)
 
 这种情况下的用户需求：
-- sdk下沉；
+
+- sdk 下沉；
 - 支持多语言；
 - 多云部署（只不过需要中间件团队自己为多云环境开发组件，没有社区现成的组件拿来用了）
 
-2. 公司有新API需求，适合做到开源项目里，于是提需求给社区，但是社区很难达成共识、争了几个月还没落地（例如https://github.com/dapr/dapr/issues/2988 ）。这种情况公司可能有业务压力，没法等那么久，希望自己先实现、落地，等社区实现新功能后再迁移过来。
+2. 公司有新 API 需求，适合做到开源项目里，于是提需求给社区，但是社区很难达成共识、争了几个月还没落地（例如https://github.com/dapr/dapr/issues/2988
+   ）。这种情况公司可能有业务压力，没法等那么久，希望自己先实现、落地，等社区实现新功能后再迁移过来。
 
 这种情况下的用户需求：
+
 - 用户对该功能自主可控，不需要（同时用中文和英文）说服社区、说服世界才能做这个功能
 - 快速扩展、服务业务
 
-3. 用户想给Dapr API加字段，先在自己的Fork版本里添加了字段、满足线上需求，然后将PR提给社区。社区拒绝添加该字段，PR被关闭。用户很尴尬：这字段已经在线上使用了，怎么处理？
+3. 用户想给 Dapr API 加字段，先在自己的 Fork 版本里添加了字段、满足线上需求，然后将 PR 提给社区。社区拒绝添加该字段，PR 被关闭。用户很尴尬：这字段已经在线上使用了，怎么处理？
 
 ## 2. High level design
+
 ### 2.1. Hierarchical API
-参考OS领域当年是怎么定API的，我们可以把Runtime API设计成多层：
 
+参考 OS 领域当年是怎么定 API 的，我们可以把 Runtime API 设计成多层：
 
 ![img.png](../../../img/api_plugin/img.png)
```

Comment:
image err

Reply:
I think it's a flaw of github preview feature. The doc site is ok:
![image](https://user-images.githubusercontent.com/26001097/159421157-940819ed-5e3d-490d-8393-703f9c1e3dd0.png)

Anyway, I will upload this image to a CDN

Reply:
@stulzq fixed.Please review again